### PR TITLE
31 removing contract data from admin page

### LINKED ISSUE #31: Removing contract data from admin page
## Description
As an admin I want to be able to remove contracts from the admin page because it is easier.

## Acceptance criteria
- [x] An admin should be able to remove a contract from the list
- [x] When removed from the list, a contract should be removed from the website 

## Dependencies
- #24

### PR DESCRIPTION
**Pull request by JacobBredin**
## Summary
Add a remove button for each contract inside the admin page.

image

## Definition of done
- [x] All acceptance criteria are completed.
- [x] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [ ] The code is reviewed and accepted by (at least) two other team members.
- [x] Views adhere to design style.

Closes #31

### CODE REVIEW

**Review by martinjonsson01 — CHANGES_REQUESTED** [review]
Great job! 

The only thing I want you to actually change is the commented out test. 
The rest are nit-picks and can be ignored if you choose to.

File: tests/Presentation.Tests/Server/Controllers/ContractsControllerTests.cs [hunk 1]
```diff
@@ -28,4 +29,46 @@ public void Get_AllContracts()
...
+
+        // Act
+        IActionResult actual = _cut.Remove(id);
+
+        // Assert
+        actual.Should().BeOfType<OkResult>();
+    }
+
+    [Fact]
+    public void Remove_CallContractService_WhenIDExisting()
```
[the remark this answers is not in the document]
Comment (martinjonsson01) [reply tests/Presentation.Tests/Server/Controllers/ContractsControllerTests.cs:48]: and `CallContractService` -> `CallsContractService`

File: src/Client/Pages/Admin/ContractListItem.razor [hunk 2]
```diff
@@ -1,12 +1,35 @@
-﻿<div id="@Name" class="list-group-item" style="max-width: 50%"> @Name </div>
+﻿@using Domain.Contracts
+@inject HttpClient _http
+
+<div id="@Contract.Name" class="list-group-item d-flex justify-content-between" style="max-width: 50%">
+     @Contract.Name
+
+     <button type="button" class="btn btn-danger" @onclick="RemoveContract">
+         Ta bort
+     </button>
```
[the remark this answers is not in the document]
Comment (JacobBredin) [reply src/Client/Pages/Admin/ContractListItem.razor:9]: Added 🙌

File: src/Infrastructure/Contracts/FakeContractRepository.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ namespace Infrastructure.Contracts;
 /// </summary>
 public class FakeContractRepository : IContractRepository
 {
-    private readonly ICollection<Contract> _contracts;
+    private readonly List<Contract> _contracts;
```
[the remark this answers is not in the document]
Comment (JacobBredin) [reply src/Infrastructure/Contracts/FakeContractRepository.cs:12]: Previously I needed it for the Find method, but now I need for the RemoveAll method. Neither ICollection or IList has RemoveAll 😢

[the remark this answers is not in the document]
Comment (martinjonsson01) [reply src/Infrastructure/Contracts/FakeContractRepository.cs:12]: Ah, that makes sense! 👍

File: tests/Application.Tests/Contracts/ContractServiceTests.cs [hunk 4]
```diff
@@ -59,4 +59,34 @@ public void AddingContract_DoesNotThrow_IfIDIsUnique()
...
+        _mockRepo.Setup(repository => repository.All).Returns(new List<Contract>());
+
+        var id = Guid.NewGuid();
+
+        // Act
+        bool actual = _cut.Remove(id);
+
+        // // Assert
+        // actual.Should().BeTrue();
+    }
```
[the remark this answers is not in the document]
Comment (JacobBredin) [reply tests/Application.Tests/Contracts/ContractServiceTests.cs:76]: Both fixed 👍

File: tests/Presentation.Tests/Server/Controllers/ContractsControllerTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JacobBredin) [reply tests/Presentation.Tests/Server/Controllers/ContractsControllerTests.cs:48]: damn, deally makes you think 😔

File: tests/Presentation.Tests/Client/Pages/Admin/ContractListTests.cs [hunk 5]
```diff
@@ -58,4 +59,27 @@ public void AddingContract_DoesNotThrow_BeforeContractsAreFetched()
...
+        MockHttp.When("/api/v1/Contracts/All").RespondJson(contracts);
+        MockHttp.When(HttpMethod.Delete, $"/api/v1/Contracts/{firstContract.Id}").Respond(req => new HttpResponseMessage(HttpStatusCode.OK));
+
+        IRenderedComponent<ContractList> cut = Context.RenderComponent<ContractList>();
+        const string itemSelector = ".list-group-item";
+        cut.WaitForElement(".btn.btn-danger");
+
+        // Act
+        await cut.Find(".btn.btn-danger").ClickAsync(new MouseEventArgs()).ConfigureAwait(false);
+        cut.WaitForState(() => cut.FindAll(".btn.btn-danger").Count == 1);
```
[the remark this answers is not in the document]
Comment (JacobBredin) [reply tests/Presentation.Tests/Client/Pages/Admin/ContractListTests.cs:78]: Thank you

File: tests/Application.Tests/Contracts/ContractServiceTests.cs [hunk 6]
```diff
@@ -59,4 +59,33 @@ public void AddingContract_DoesNotThrow_IfIDIsUnique()
...
+        // Arrange
+        var contract = new Contract();
+        _mockRepo.Setup(repository => repository.Remove(contract.Id)).Returns(true);
+
+        // Act
+        bool actual = _cut.Remove(contract.Id);
+
+        // // Assert
+        actual.Should().BeTrue();
+    }
```
Comment (martinjonsson01) [inline tests/Application.Tests/Contracts/ContractServiceTests.cs:75]: nit:
The test name says `RemovingContractFromRepo` even though what's being tested here is the `ContractService`, so a better name would be `RemovingContract_ReturnsTrue_WhenContractExists`

File: tests/Application.Tests/Contracts/ContractServiceTests.cs [hunk 7]
```diff
@@ -59,4 +59,33 @@ public void AddingContract_DoesNotThrow_IfIDIsUnique()
...
+
+    [Fact]
+    public void RemovingContractFrom_DoesReturnFalse_WhenRepoIsEmpty()
```
Comment (martinjonsson01) [inline tests/Application.Tests/Contracts/ContractServiceTests.cs:78]: nit:
I think you missed the part after `From` here, but like the previous comment you can just call it  `RemovingContract_ReturnsFalse_WhenRepoIsEmpty`

**Review by martinjonsson01 — APPROVED** [review]
lgtm 👍

File: tests/Application.Tests/Contracts/ContractServiceTests.cs (hunk 6 above)
Comment (JacobBredin) [reply tests/Application.Tests/Contracts/ContractServiceTests.cs]: Fixed, I hope 👍

**Review by jonathan-lindqvist — APPROVED** [review]
Nice!

**Pull request merged**